main.cpp

```diff
@@ -186,21 +208,23 @@ isDisabled(false)
 
 bool VendingMachine::ItemDispenser::distributeItems(int numberOfItems)
 {
-    if (inventory >= numberOfItems)
+    for (int i = 0; i < numberOfItems; ++i)
     {
-        inventory -= numberOfItems;
-        if (inventory == 0)
+        std::cout << "Distributing item\n";
+        --inventory;
+        if (inventory == 0) 
         {
-            disable("Empty"); 
+            disable("Empty");
+            return (++i == numberOfItems);
```

Comment:
Don't write code like this. 
It is confusing.
Does the comparison happen first, followed by the increment? 
Does the increment happen first, followed by the comparison? 

Your job as a programmer is to write code that is easy to understand FOR THE PERSON WHO WILL READ IT. 
That could be you in 6 months. 
It could be someone else.
Help yourself by writing SIMPLE, EASY TO UNDERSTAND code
- [ ] revise